src/reasoning-layer-lib/connectors-reasoning-layer.xsl

```diff
@@ -198,32 +216,56 @@
     <xd:doc>
         <xd:desc>Rule 20 (Association inverse in reasnoning layer) . Specify inverse object property
             between the source and target ends of the association.</xd:desc>
-        <xd:param name="connector"/>
+        <xd:param name="connectorName"/>
+        <xd:param name="root"/>
     </xd:doc>
 
     <xsl:template name="connectorInverse">
-        <xsl:param name="connector"/>
-        <xsl:if test="$connector/properties/@direction = 'Bi-Directional'">
-            <xsl:variable name="targetRole"
-                select="
-                    if (boolean($connector/target/role/@name)) then
-                        f:lexicalQNameToWords($connector/target/role/@name)
-                    else
-                        concat($mockUnknownPrefix, ':', $mockUnnamedElement)"/>
-            <xsl:variable name="targetRoleURI"
-                select="f:buildURIfromLexicalQName($targetRole, fn:false())"/>
-            <xsl:variable name="sourceRole"
-                select="
-                    if (boolean($connector/source/role/@name)) then
-                        f:lexicalQNameToWords($connector/source/role/@name)
-                    else
-                        concat($mockUnknownPrefix, ':', $mockUnnamedElement)"/>
-            <xsl:variable name="sourceRoleURI"
-                select="f:buildURIfromLexicalQName($sourceRole, fn:false())"/>
-            <owl:ObjectProperty rdf:about="{$targetRoleURI}">
-                <owl:inverseOf rdf:resource="{$sourceRoleURI}"/>
-            </owl:ObjectProperty>
-        </xsl:if>
+        <xsl:param name="connectorName"/>
+        <xsl:param name="root"/>
+        <xsl:variable name="connectorsWithSameName"
+            select="f:getConnectorByName($connectorName, $root)"/>
+        <xsl:variable name="bidirectionalConnectors"
```

Comment:
what about the bidirectional connectors for which the $connectorName == source/role/@name? Are we missing these cases?